js/Snake.js

```diff
@@ -64,40 +57,42 @@ class Snake extends Drawable {
 
     if (this.head.position.x < 0 || this.head.position.x >= width) {
       return true;
-      
+
     } else if (this.head.position.y < 0 || this.head.position.y >= height) {
       return true;
 
-    } else if (isSnake.find(b => b.eql(this.head.position))) {
+    } else if (isSnake.find((b) => b.eql(this.head.position))) {
       return true;
 
     } else {
       return false;
     }
   }
 
+  isOppositeDirection(direction) {
+    const opposite = {
+      left: 'right',
+      right: 'left',
+      up: 'down',
+      down: 'up',
+    };
+
+    return direction === opposite[this.direction];
+  }
+
   move() {
     if (this.growLeft > 0) {
+      const bodyPart = new SnakeBody(
```

Comment:
What about someting like

```js
const bodyPart = new SnakeBody(this.body[this.body.length - 1].clone());
```

Reply:
Is `clone` a native method? I don't know about it. (I will do my own research)

Reply:
Nope